gensim/models/word2vec.py

```diff
@@ -1191,383 +1191,78 @@ def intersect_word2vec_format(self, fname, lockf=0.0, binary=False, encoding='ut
-        only check the first 10000 word vectors in the vocabulary order. (This may be
-        meaningful if you've sorted the vocabulary by descending frequency.)
-
-        Example::
-
-          >>> trained_model.similar_by_vector([1,2])
-          [('survey', 0.9942699074745178), ...]
-
-        """
-
-        return self.most_similar(positive=[vector], topn=topn, restrict_vocab=restrict_vocab)
+        return self.kv.similar_by_vector(vector, topn, restrict_vocab)
 
     def doesnt_match(self, words):
-        """
-        Which word from the given list doesn't go with the others?
-
-        Example::
-
-          >>> trained_model.doesnt_match("breakfast cereal dinner lunch".split())
-          'cereal'
-
-        """
-        self.init_sims()
-
-        words = [word for word in words if word in self.vocab]  # filter out OOV words
-        logger.debug("using words %s" % words)
-        if not words:
-            raise ValueError("cannot select a word from an empty list")
-        vectors = vstack(self.syn0norm[self.vocab[word].index] for word in words).astype(REAL)
-        mean = matutils.unitvec(vectors.mean(axis=0)).astype(REAL)
-        dists = dot(vectors, mean)
-        return sorted(zip(dists, words))[0][1]
+        return self.kv.doesnt_match(words)
 
     def __getitem__(self, words):
+        return self.kv.__getitem__(words)
 
+    def __getattr__(self, item):
         """
-        Accept a single word or a list of words as input.
-
-        If a single word: returns the word's representations in vector space, as
-        a 1D numpy array.
-
-        Multiple words: return the words' representations in vector space, as a
-        2d numpy array: #words x #vector_size. Matrix rows are in the same order
-        as in input.
-
-        Example::
-
-          >>> trained_model['office']
-          array([ -1.40128313e-02, ...])
-
-          >>> trained_model[['office', 'products']]
-          array([ -1.40128313e-02, ...]
-                [ -1.70425311e-03, ...]
-                 ...)
-
+        To maintain backwards compatibility, calls such as trained_model.syn0norm won't break but will be
+        rerouted to trained_model.kv.syn0norm
         """
-        if isinstance(words, string_types):
-            # allow calls like trained_model['office'], as a shorthand for trained_model[['office']]
-            return self.syn0[self.vocab[words].index]
-
-        return vstack([self.syn0[self.vocab[word].index] for word in words])
+        if item == "syn0":
+            return self.kv.syn0
+        if item == "syn0norm":
+            return self.kv.syn0norm
+        if item == "vocab":
+            return self.kv.vocab
+        if item == "index2word":
+            return self.kv.index2word
+        raise AttributeError("'{0}' object has no attribute '{1}'".format(type(self).__name__, item))
 
     def __contains__(self, word):
-        return word in self.vocab
+        return self.kv.__contains__(word)
```

Comment:
Explicitly calling `__contains__` seems a little awkward. `return word in self.kv`?